src/functionaltest/java/com/ericsson/ei/config/SeleniumConfig.java

```diff
@@ -1,44 +1,77 @@
 package com.ericsson.ei.config;
 
 import com.ericsson.ei.frontend.exception.OSNotSupportedException;
+import com.ericsson.ei.frontend.exception.PropertiesNotLoadedException;
 import com.google.common.io.Files;
 import org.apache.commons.lang3.SystemUtils;
+import org.openqa.selenium.firefox.FirefoxBinary;
 import org.openqa.selenium.firefox.FirefoxDriver;
 import org.openqa.selenium.firefox.FirefoxOptions;
 import org.openqa.selenium.firefox.FirefoxProfile;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Properties;
 
 import java.io.File;
 
 public class SeleniumConfig {
-
     private static final Logger LOGGER = LoggerFactory.getLogger(SeleniumConfig.class);
 
+    private static String propertiesPath = String.join(File.separator, "src", "functionaltest", "resources");
+    private static String propertiesFile = "functional-test.properties";
+
+    private static String firefoxBZip2FileUrlLinux = "";
+    private static String firefoxBZip2FileNameLinux = "";
+
     private static File tempDownloadDirectory = Files.createTempDir();
+    private static FirefoxDriver driver;
 
-    public static FirefoxDriver getFirefoxDriver() throws OSNotSupportedException {
-        FirefoxDriver driver;
-        FirefoxProfile firefoxProfile = new FirefoxProfile();
+    public static FirefoxDriver initFirefoxDriver() throws PropertiesNotLoadedException, OSNotSupportedException {
+        if (driver != null) {
+            return driver;
+        }
 
+        FirefoxProfile firefoxProfile = new FirefoxProfile();
         firefoxProfile.setPreference("browser.download.folderList", 2);
         firefoxProfile.setPreference("browser.download.dir", tempDownloadDirectory.getPath());
         firefoxProfile.setPreference("browser.helperApps.neverAsk.saveToDisk", "application/json");
 
+        FirefoxOptions firefoxOptions = new FirefoxOptions()
+                .setHeadless(true)
+                .setProfile(firefoxProfile);
+
+        boolean successfullyLoadedProperties = loadProperties();
+        if (!successfullyLoadedProperties) {
```

Comment:
simplify this ->if (!loadProperties())

Reply:
I think it's simpler to write it your way but not as descriptive when reading the code. When calling functions in a conditional statement i think it's way clearer to declare a variable with a name that explains the returned value even in a simple case as this. If i'm the only one who thinks that, i'll change it of course.  :)

Reply:
Yes it could be simplified, but i think that declaring a variable before with the result from the method makes it way more clearer in a conditional statement such as if-statement even in a simple case such as this one. Don't you agree?